src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2744,168 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
+            logging.warning(f"Error: {line} - {e}")    
+            continue
+
+        md[k.lower()] = v
+    return md
+
+
+def _convert_tescan_to_odemis_metadata(metadata: str) -> dict:
+    """
+    Converts TESCAN metadata to odemis metadata format
+    :param metadata: metadata string as ini (TESCAN format)
+    :return: (dict) metadata as dictionary (odemis format)
+    :raises: (Exception): if the metadata cannot be parsed
+    """
+
+    # NOTE: due to metadata corruption, we can't directly read the beam type
+    # the metadata should be in ini format with two sections, MAIN and SEM/FIB
+    # the SEM/FIB section header allows us to determine the beam type, 
+    # but it is corrupt and not parsed correctly by configparser. 
+    # Therefore we need to parse the metadata manually 
+    # and can't determine the beam type
+    # Once fixed:
+    # 1. Change to configparser to parse the metadata correctly
+    # 2. Correctly read the beam type from section headers
+    # 3. Remove the beam type check below (acq_type)
+
+    # parse ini metadata as dictionary 
+    # NOTE: not correctly ini formatted due to corruption
+    try:
+       tmd = _parse_tescan_metadata(metadata)
+    except Exception as e:
+        raise Exception(f"Failed to parse TESCAN metadata: {e}")
+
+    md = {}
+    
+    # tmp: inferred from metadata
+    beam_type = None
+
+    # general
+    try:
+        datetime_str = tmd["date"] + " " + tmd["time"]
+        date = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_ACQ_DATE} - {e}")
+
+    try:
+        md[model.MD_HW_NAME] = tmd["device"] + "-" + tmd["serialnumber"]    
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {e}")
+    try:
+        md[model.MD_HW_VERSION] = tmd["device"] + "-" + tmd["devicemodel"]    
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {e}")
+    
+    try:
+        md[model.MD_SW_VERSION] = tmd["softwareversion"]
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_SW_VERSION} - {e}")
+
+    try:
+        md[model.MD_PIXEL_SIZE] = (float(tmd["pixelsizex"]), 
+                                   float(tmd["pixelsizey"]))
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_PIXEL_SIZE} - {e}")
+
+    try:
+        md[model.MD_DWELL_TIME] = float(tmd["dwelltime"])
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_DWELL_TIME} - {e}")
+    try:
+        # MD_POS should match image dimensions, e.g. only XY for 2D, XYZ for 3D, etc.
+        md[model.MD_POS] = (float(tmd["stagex"]), 
+                           float(tmd["stagey"]),
+                           )
+    except Exception as e:
```

Comment:
I think in all the exceptions from line 2827 to here you can use a KeyError instead of a generic Exception: `except KeyError as e:`

Reply:
same from line 2878 to line 2909